### ISSUE TITLE
toHaveBeenCalledBefore throw an error

### ISSUE THREAD

**Issue by nolazybits** [jest-community/jest-extended#147]
**Bug**

- `package` version: 0.7.2
- `node` version: 10
- `npm` (or `yarn`) version: 1.7

From the documentation example

```javascript
describe("toHaveBeenCalledBefore", async () =>
{
    it('calls mock1 before mock2', () => {
      const mock1 = jest.fn();
      const mock2 = jest.fn();

      mock1();
      mock2();
      mock1();

      expect(mock1).toHaveBeenCalledBefore(mock2);
    });
}
```
you will get the following error

```
   TypeError: Cannot read property 'length' of undefined

      170 |         mock1();
      171 | 
    > 172 |         expect(mock1).toHaveBeenCalledBefore(mock2);
          |                       ^
      173 |     });
      174 | 

      at Object.<anonymous>.exports.default (../../node_modules/jest-extended/dist/matchers/toHaveBeenCalledBefore/predicate.js:9:23)
      at Object.toHaveBeenCalledBefore (../../node_modules/jest-extended/dist/matchers/toHaveBeenCalledBefore/index.js:23:42)
      at Object.<anonymous>.it (tests/unit/BugReport.test.ts:172:23)
```

**Comment by benjaminkay93** [comment]
Your using jest 23 right? this has been fixed in a PR that been merged in https://github.com/jest-community/jest-extended/pull/132, However this has not yet been published, until then you will have to use either jest 22, or not use this matcher :P

**Comment by benjaminkay93** [comment]
However the docs on here do say jest 22=<  so its expected to be working in jest 23, I know the docs have been updated in https://github.com/jest-community/jest-extended/pull/132 but is it worth keeping them compatible with 22 as well?

**Comment by nolazybits** [comment]
Awesome thank you much for the feedback.
will keep an eye on the PR

Came up with a bad workaround
```typescript
...
        const moduleOrder: number[] = [];
        const sut: ISauceCore = await Container.GetFromContainer(SauceCore).isReady;

        const spyModuleStart: SpyInstance = jest.spyOn(Module.prototype, "start");
        const spyModule2Start: SpyInstance = jest.spyOn(Module2.prototype, "start");
        const spyModule3Start: SpyInstance = jest.spyOn(Module3.prototype, "start");
        spyModuleStart.mockImplementation(async () =>
        {
            moduleOrder.push(1);
        });
        spyModule2Start.mockImplementation(async () =>
        {
            moduleOrder.push(2);
        });
        spyModule3Start.mockImplementation(async () =>
        {
            moduleOrder.push(3);
        });

        await sut.register(Module);
        await sut.register(Module2);
        await sut.register(Module3);
        await sut.start();

        expect(moduleOrder).toEqual(expect.arrayContaining([1, 2, 3]));

        spyModuleStart.mockRestore();
        spyModule2Start.mockRestore();
        spyModule3Start.mockRestore();
...
```
mocking my function and pushing to an array and checking the array against the expected :/

**Comment by benjaminkay93** [comment]
you could do 
``` javascript
const mock1 = jest.fn()
const mock2 = jest.fn()

expect(mock1.mock. invocationCallOrder).toBeLessThan(mock2.mock.invocationCallOrder)
```

**Comment by benjaminkay93** [comment]
The jest api changed from timestamp to invocationCallOrder so you dont have to mock out a timeout function inside the jest.fn() to get the innvocation order.

**Comment by nolazybits** [comment]
Thank you for the feedback
I'll give it a try on Monday :)

**Comment by jonohayon** [comment]
Having the same problem, can the module owner publish a new version to npm?

**Comment by mattphillips** [comment]
Hey thanks for raising this, sorry about the delay in getting this published! It is now out in [v0.8.0](https://www.npmjs.com/package/jest-extended/v/0.8.0)
